vms/platformvm/state/stakers_crud_ops_test.go

```diff
@@ -0,0 +1,535 @@
+				expectedResult := []*Staker{}
+				require.NoError(checkDelegatorsContent(store, pending, delegator.SubnetID, delegator.NodeID, expectedResult))
+
+				// range query
+				require.NoError(checkStakersContent(store, pending, expectedResult))
+			}
+
+			{
+				// it's fine deleting unknown delegator
+				store.DeletePendingDelegator(delegator)
+
+				// query
+				expectedResult := []*Staker{}
+				require.NoError(checkDelegatorsContent(store, pending, delegator.SubnetID, delegator.NodeID, expectedResult))
+
+				// range query
+				require.NoError(checkStakersContent(store, pending, expectedResult))
+			}
+
+			{
+				// insert the delegator and show it can be found
+				store.PutPendingDelegator(delegator)
+
+				// query
+				expectedResult := []*Staker{delegator}
+				require.NoError(checkDelegatorsContent(store, pending, delegator.SubnetID, delegator.NodeID, expectedResult))
+
+				// range query
+				require.NoError(checkStakersContent(store, pending, expectedResult))
+			}
+
+			{
+				// delete the delegator and show it's not found anymore
+				store.DeletePendingDelegator(delegator)
+
+				// query
+				expectedResult := []*Staker{}
+				require.NoError(checkDelegatorsContent(store, pending, delegator.SubnetID, delegator.NodeID, expectedResult))
+
+				// range query
+				require.NoError(checkStakersContent(store, pending, expectedResult))
+			}
+		})
+	}
+}
+
+// [checkStakersContent] verifies whether store contains exactly the stakers specified in the list.
+// stakers order does not matter. stakers slice gets consumed while checking.
+func checkStakersContent(store Stakers, validatorsType uint8, expectedValidators []*Staker) error {
+	var (
+		it  StakerIterator
+		err error
+	)
+
+	switch validatorsType {
+	case current:
+		it, err = store.GetCurrentStakerIterator()
+	case pending:
+		it, err = store.GetPendingStakerIterator()
+	default:
+		return errors.New("Unhandled stakers status")
+	}
+	if err != nil {
+		return fmt.Errorf("unexpected failure in staker iterator creation, error %w", err)
+	}
+	defer it.Release()
+
+	if len(expectedValidators) == 0 {
+		if it.Next() {
+			return fmt.Errorf("expected empty iterator, got at least element %v", it.Value())
+		}
+		return nil
+	}
+
+	for it.Next() {
+		var (
+			staker = it.Value()
+			found  = false
+
+			retrievedStakerIdx = 0
+		)
+
+		for idx, s := range expectedValidators {
+			if reflect.DeepEqual(staker, s) {
+				retrievedStakerIdx = idx
+				found = true
+			}
+		}
+		if !found {
+			return fmt.Errorf("found extra staker %v", staker)
+		}
+		expectedValidators[retrievedStakerIdx] = expectedValidators[len(expectedValidators)-1] // order does not matter
+		expectedValidators = expectedValidators[:len(expectedValidators)-1]
+	}
+
+	if len(expectedValidators) != 0 {
+		return errors.New("missing stakers")
+	}
+	return nil
+}
+
+func checkDelegatorsContent(store Stakers, delegatorsType uint8, subnetID ids.ID, nodeID ids.NodeID, expectedDelegators []*Staker) error {
```

Comment:
Same about passing a `require` in. Also comment like `checkStakersContent` would be good.

Reply:
Done